### PR TITLE
feat: add automatic signature downloader

### PR DESCRIPTION
**Pull request by josesilveiraa**
## What does this PR adds?
- An automatic signature downloader from ReVanced repos, allowing to build the modified app easily.
- It also refactors `util` package.

### CODE REVIEW

**Comment by josesilveiraa** [comment]
The changes weren't tested because of branch conflicts in gradle.

**Comment by josesilveiraa** [comment]
oh i mispelt downloader

File: src/main/kotlin/app/revanced/cli/Main.kt [hunk 1]
```diff
@@ -129,6 +134,12 @@ class Main {
                 shortName = "o",
                 description = "Output directory"
             ).required()
+            val downloadSignatures by parser.option(
+                ArgType.Boolean,
+                fullName = "downloadSignatures",
```
Comment (Lucaskyy) [inline src/main/kotlin/app/revanced/cli/Main.kt:139]: Should be kebab-case.
```suggestion
                fullName = "download-signatures",
```

File: src/main/kotlin/app/revanced/cli/utils/signature/SignatureDownloader.kt [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+import java.io.File
+import java.net.URL
+import java.nio.file.Files
+import java.nio.file.Paths
+
+object SignatureDownloader {
+
+    fun download(out: String) {
+        val url =
+            URL("https://raw.githubusercontent.com/ReVancedTeam/revanced-signatures/main/signatures/youtube.signatures.json")
```
Comment (Lucaskyy) [inline src/main/kotlin/app/revanced/cli/utils/signature/SignatureDownloader.kt:12]: This is the wrong branch for now (until PRs are merged), but you can keep it like this.

**Review by Lucaskyy — CHANGES_REQUESTED** [review]
Besides the 2 changes below, please also make the `signatures` argument optional (remove `.required()` and make type nullable, see `integrations` argument as a reference).
Lastly, please reformat the code (using IntelliJ is preferred).

File: src/main/kotlin/app/revanced/cli/utils/signature/SignatureDownloader.kt (hunk 2 above)
Comment (josesilveiraa) [reply src/main/kotlin/app/revanced/cli/utils/signature/SignatureDownloader.kt:12]: i'll take a look

Comment (JohnShiozo) [reply src/main/kotlin/app/revanced/cli/utils/signature/SignatureDownloader.kt:12]: epic

**Comment by Lucaskyy** [comment]
This PR is no longer needed since the Patcher was modified to develop signatures together with the Patches. Thank you anyway though for the contribution!

**Pull request closed without merging**